lib/src/builder/parsing/ast_util.dart

```diff
@@ -42,17 +42,16 @@ extension InitializerHelperTopLevel on TopLevelVariableDeclaration {
   /// The first variable in this list.
   VariableDeclaration get firstVariable => variables.firstVariable;
 
-  /// Returns whether or not the config argument of `uiFunction` is generated.
-  bool get hasGeneratedConfigArg {
+  /// Returns whether or not the config argument of a Function type is generated.
+  bool get usesAGeneratedConfig {
     return firstInitializer != null &&
         anyDescendantIdentifiers(firstInitializer, (identifier) {
-          if (identifier.name == 'uiFunction') {
-            final args =
-                identifier.thisOrAncestorOfType<MethodInvocation>()?.argumentList?.arguments;
-            if (args == null || args.length < 2) return false;
-            return args[1].toString().startsWith(RegExp(r'\$'));
-          }
-          return false;
+          final uiFactoryDeclaration = identifier.thisOrAncestorOfType<VariableDeclaration>();
+
+          return firstInitializer != null &&
+              anyDescendantIdentifiers(firstInitializer, (identifier) {
```

Comment:
This is already inside a `firstInitializer != null` and `anyDescendantIdentifiers` check